domain-server/resources/web/web-new/src/boot/axios.ts

```diff
@@ -0,0 +1,45 @@
+//  axios.ts
+//
+//  Created by Kalila L. on Sep. 4th, 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+import { boot } from "quasar/wrappers";
+import axios, { AxiosInstance } from "axios";
+import Log from "../modules/utilities/log";
+
+declare module "@vue/runtime-core" {
+    interface ComponentCustomProperties {
+        $axios: AxiosInstance;
+    }
+}
+
+Log.info(Log.types.OTHER, "Bootstrapping Axios.");
+
+axios.defaults.headers.common = {
```

Comment:
Would they really refuse it? That's kind off odd to refuse something with a bonus header. Either way, we'll probably have to adjust how this is handled later then. Probably with something like `beforeSend` if Axios has something similar to that.

Reply:
HTTP is very picky these days. Requests are preceded with and OPTIONS request which asks which header are accepted. The server must specify the "allowed headers" -- even ones like "Authorization" have to specified.
This is why I'd suggest creating some utility routines like "doGET", "doPOST", etc to centralize the needed parameters and error processing rather than having raw calls to axios scattered around.